### PR TITLE
Test: Replace t.error/fatal with assert/request in [rawnode_test.go]

### PR DESCRIPTION
**Pull request by MrDXY**
Refactor: Replacing t.Error/t.Fatal with assert/require

This commit tries to eliminates boilerplate code in the testing suite by replacing occurrences
of t.Error and t.Fatal with assertions from the assert and require packages.

Changes Made:

- Replaced instances of t.Error with assert
- Replaced instances of t.Fatal with require

Affected Files:
- [rawnode_test.go]

Part of: #146

### CODE REVIEW

File: rawnode_test.go [hunk 1]
```diff
@@ -224,13 +219,11 @@ func TestRawNodeProposeAndConfChange(t *testing.T) {
...
 
-	for _, tc := range testCases {
+	for i, tc := range testCases {
 		t.Run("", func(t *testing.T) {
 			s := newTestMemoryStorage(withPeers(1))
 			rawNode, err := NewRawNode(newTestConfig(1, 10, 1, s))
-			if err != nil {
-				t.Fatal(err)
-			}
+			require.NoError(t, err, "#%d", i)
```
Comment (pav-kv) [inline rawnode_test.go:226]: This `#%d` printing is unnecessary. The purpose of this printing is to be able to identify which subtest/iteration failed when debugging test failures. However, in this case, every subtest is wrapped in `t.Run()` which makes it a separate test with a unique name. In this case, the test name will be an incremental index. If a subtest fails, the error message will contain this subtest name.

File: rawnode_test.go [hunk 2]
```diff
@@ -644,15 +547,9 @@ func TestRawNodeReadIndex(t *testing.T) {
...
-	}
-	if msgs[0].Type != pb.MsgReadIndex {
-		t.Errorf("msg type = %d, want %d", msgs[0].Type, pb.MsgReadIndex)
-	}
-	if !bytes.Equal(msgs[0].Entries[0].Data, wrequestCtx) {
-		t.Errorf("data = %v, want %v", msgs[0].Entries[0].Data, wrequestCtx)
-	}
+	require.Len(t, msgs, 1)
+	assert.Equal(t, pb.MsgReadIndex, msgs[0].Type)
+	assert.Equal(t, msgs[0].Entries[0].Data, wrequestCtx)
```
Comment (pav-kv) [inline rawnode_test.go:552]: Swap `msgs[0].Entries[0].Data` and `wrequestCtx`. First goes the "expected" value, then the "actual".

**Review by pav-kv — APPROVED** [review]
LGTM % a couple of nits.

**Comment by ahrtr** [comment]
Please rebase this PR

**Review by ahrtr — APPROVED** [review]
lgtm

Thanks

**Pull request merged**